icu4c/source/i18n/number_usageprefs.cpp

```diff
@@ -0,0 +1,233 @@
+    if (U_FAILURE(status)) {
+        return;
+    }
+
+    quantity.roundToInfinity(); // Enables toDouble
+    const auto routed = fUnitsRouter.route(quantity.toDouble(), status);
+    if (U_FAILURE(status)) {
+        return;
+    }
+    const MaybeStackVector<Measure>& routedUnits = routed.measures;
+    micros.outputUnit = routed.outputUnit.copy(status).build(status);
+    if (U_FAILURE(status)) {
+        return;
+    }
+
+    mixedMeasuresToMicros(routedUnits, &quantity, &micros, status);
+
+    UnicodeString precisionSkeleton = routed.precision;
+    if (micros.rounder.fPrecision.isBogus()) {
+        if (precisionSkeleton.length() > 0) {
+            micros.rounder.fPrecision = parseSkeletonToPrecision(precisionSkeleton, status);
+        } else {
+            // We use the same rounding mode as COMPACT notation: known to be a
+            // human-friendly rounding mode: integers, but add a decimal digit
+            // as needed to ensure we have at least 2 significant digits.
+            micros.rounder.fPrecision = Precision::integer().withMinDigits(2);
+        }
+    }
+}
+
+Precision UsagePrefsHandler::parseSkeletonToPrecision(icu::UnicodeString precisionSkeleton,
+                                                      UErrorCode status) {
+    if (U_FAILURE(status)) {
+        // As a member of UsagePrefsHandler, which is a friend of Precision, we
+        // get access to the default constructor.
+        return {};
+    }
+    constexpr int32_t kSkelPrefixLen = 20;
+    if (!precisionSkeleton.startsWith(UNICODE_STRING_SIMPLE("precision-increment/"))) {
+        status = U_INVALID_FORMAT_ERROR;
+        return {};
+    }
+    U_ASSERT(precisionSkeleton[kSkelPrefixLen - 1] == u'/');
+    StringSegment segment(precisionSkeleton, false);
+    segment.adjustOffset(kSkelPrefixLen);
+    MacroProps macros;
+    blueprint_helpers::parseIncrementOption(segment, macros, status);
+    return macros.precision;
+}
+
+UnitConversionHandler::UnitConversionHandler(const MeasureUnit &unit, const MicroPropsGenerator *parent,
+                                             UErrorCode &status)
+    : fOutputUnit(unit), fParent(parent) {
+    MeasureUnitImpl temp;
+    const MeasureUnitImpl &outputUnit = MeasureUnitImpl::forMeasureUnit(unit, temp, status);
+    const MeasureUnitImpl *inputUnit = &outputUnit;
+    MaybeStackVector<MeasureUnitImpl> singleUnits;
+    U_ASSERT(outputUnit.complexity == UMEASURE_UNIT_MIXED);
+    // When we wish to support unit conversion, replace the above assert with this if:
+    // if (outputUnit.complexity == UMEASURE_UNIT_MIXED) {
+    {
+        singleUnits = outputUnit.extractIndividualUnits(status);
+        U_ASSERT(singleUnits.length() > 0);
+        inputUnit = singleUnits[0];
+    }
+    // TODO: this should become an initOnce thing? Review with other
+    // ConversionRates usages.
+    ConversionRates conversionRates(status);
+    if (U_FAILURE(status)) {
+        return;
+    }
```

Comment:
We should pass the conversionRates.

Reply:
The question for the TODO is from where though. Where is the right place to load the conversion rates in the first place? Once we know this, we can figure out how to plumb it through in the neatest fashion.